pkg/cloudprovider/cloud.go

```diff
@@ -66,7 +66,8 @@ type TCPLoadBalancer interface {
 	CreateTCPLoadBalancer(name, region string, externalIP net.IP, ports []int, hosts []string, affinityType api.AffinityType) (string, error)
 	// UpdateTCPLoadBalancer updates hosts under the specified load balancer.
 	UpdateTCPLoadBalancer(name, region string, hosts []string) error
-	// DeleteTCPLoadBalancer deletes a specified load balancer.
+	// DeleteTCPLoadBalancer deletes a specified load balancer. It should return
+	// nil if the load balancer specified already didn't exist.
 	DeleteTCPLoadBalancer(name, region string) error
```

Comment:
I'd suggest renaming to something like EnsureTCPLoadBalancerDeleted() to better advertise it's idempotence.